### ISSUE TITLE
TypeError: Cannot read property 'length' of undefined

### ISSUE THREAD

**Issue by christroutner** [orbitdb/orbit-db#839]
I'm getting a new error with OrbitDB that I've never seen before. The code base hasn't changed. The only thing that has changed is a fresh `npm install`.

- The error occurs when trying to use [this OrbitDB-based chat example](https://github.com/christroutner/e2e-orbitDB-chat-app-prototype)

Here is the error message:
```
TypeError: Cannot read property 'length' of undefined
    at Function.concat (buffer.js:543:19)
    at tagCID (/home/trout/tmp/e2e-orbitDB-chat-app-prototype/ChatMaster-Node/node_modules/ipld/node_modules/ipld-dag-cbor/src/util.js:20:47)
    at transform (/home/trout/tmp/e2e-orbitDB-chat-app-prototype/ChatMaster-Node/node_modules/ipld/node_modules/ipld-dag-cbor/src/util.js:47:14)
    at Array.map (<anonymous>)
    at transform (/home/trout/tmp/e2e-orbitDB-chat-app-prototype/ChatMaster-Node/node_modules/ipld/node_modules/ipld-dag-cbor/src/util.js:43:18)
    at /home/trout/tmp/e2e-orbitDB-chat-app-prototype/ChatMaster-Node/node_modules/ipld/node_modules/ipld-dag-cbor/src/util.js:57:22
    at Array.forEach (<anonymous>)
    at transform (/home/trout/tmp/e2e-orbitDB-chat-app-prototype/ChatMaster-Node/node_modules/ipld/node_modules/ipld-dag-cbor/src/util.js:55:12)
    at replaceCIDbyTAG (/home/trout/tmp/e2e-orbitDB-chat-app-prototype/ChatMaster-Node/node_modules/ipld/node_modules/ipld-dag-cbor/src/util.js:68:10)
    at Object.exports.serialize (/home/trout/tmp/e2e-orbitDB-chat-app-prototype/ChatMaster-Node/node_modules/ipld/node_modules/ipld-dag-cbor/src/util.js:135:22)
```

The error appears to originate from the [ipld-dab-cbor](https://www.npmjs.com/package/ipld-dag-cbor) package, but their code hasn't changed in a month and this error suddenly appeared around September 6th.

The chat example uses these packages:
- ipfs v0.48.2
- orbit-db v0.24.2

I tried updating the the latest versions:
- ipfs v0.50.1
- orbit-db v0.25.1

With updated versions, the app does not error out, but the OrbitDB instance never connects. It hangs forever on `await db.load()`

I'm not sure where to continue to look for the root cause of this issue. I'll post a link to this issue on the relevant GitHub repositories.

**Comment by mikeal** [comment]
@gozala do you think the CID changes somehow slipped in here?

**Comment by aphelionz** [comment]
This is more or less handled in the various dependency updates in https://github.com/orbitdb/orbit-db/pull/837

However now I'm getting timeouts in go-ipfs 0.6.0 so I'm hesitant to merge, but I could and then deal with go-ipfs in a later PR, maybe

**Comment by Gozala** [comment]
> @Gozala do you think the CID changes somehow slipped in here?

Seems like this change is picked up

https://github.com/multiformats/js-cid/pull/117

But corresponding change in dag-cbor is not

https://github.com/ipld/js-ipld-dag-cbor/commit/41587e7f73dbd5196e996ca9fbe9c9a6e5d01db9#diff-34a6d62af0cf0b784f8444529f3130ef

**Comment by phillmac** [comment]
Afaik we only have compatibility with js-ipfs v0.46.0 right now. We're working hard on a js-ipfs v0.5.0 compatible version, it should be ready for release soon barring any horrible show stopper bugs

**Comment by aphelionz** [comment]
Small correction to the above: we _officially_ support up to js-ipfs 0.49.1 and go-ipfs 0.5.1. The recently released `orbit-db@0.26.0-rc1` aka `orbit-db@next` is a stab at supporting those, but still working out some kinks

**Comment by vmx** [comment]
Steps to reproduce:

```console
$ git clone https://github.com/christroutner/e2e-orbitDB-chat-app-prototype.git
$ cd e2e-orbitDB-chat-app-prototype/ChatMaster-Node
$ npm install
$ npm start
```

Now enter e.g. `someusername`, then `foo`, then `bar`. Then you'll get the `TypeError: Cannot read property 'length' of undefined` error.

**Comment by achingbrain** [comment]
Looks like `orbit-db-io`, `ipfs-log` and `orbit-db` have been released with `cids@1.0.0` which causes the above error:

```console
$ npm ls cids
...
└─┬ orbit-db@0.24.2
  ├── cids@0.7.5 
  ├─┬ orbit-db-io@0.2.0
  │ └── cids@0.7.5 
  └─┬ orbit-db-store@3.3.1
    └─┬ ipfs-log@4.6.4
      └─┬ orbit-db-io@0.3.0
        ├── cids@1.0.0 
        └─┬ ipld-dag-pb@0.20.0
          └── cids@1.0.0  deduped
```

```console
$ find . -type d -name cids
./node_modules/cids
./node_modules/orbit-db-io/node_modules/cids
./node_modules/ipfs-log/node_modules/cids
./node_modules/orbit-db/node_modules/cids
```

If I manually remove the three deep `cids` deps, the app works as expected.

E.g. run:

```console
$ npm i
$ rm -rf node_modules/orbit-db-io/node_modules/cids node_modules/ipfs-log/node_modules/cids node_modules/orbit-db/node_modules/cids
$ npm start
```

**Comment by aphelionz** [comment]
I believe these can be closed
